llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -861,9 +861,11 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
   }
   // TODO: add more intrinsic
   case Intrinsic::experimental_stepvector: {
-    unsigned Cost = 1; // vid
     auto LT = getTypeLegalizationCost(RetTy);
-    return Cost + (LT.first - 1);
+    if (ST->hasVInstructions())
+      return LT.first *
```

Comment:
This isn't correct. If LT.first is greater than 1. Then there is 1 vid.v and LT.first-1 vadd.vx/vi instructions.

Reply:
Thanks for catching that!! Fixed.